contents/java/Spring.md

```diff
@@ -0,0 +1,277 @@
+need to use Java Database Connectivity (JDBC) API to access tabular data stored in any relational database.
+It is a part of JavaSE (Java Standard Edition), from Oracle Corporation.
+    
+In a traditional JDBC API:
+- a typical SQL query would involve quite some amount of boilerplate code, such as creating connection,
+statement, closing resultset, connection etc.
+- exception handling has to be manually written out.
+- transactions need to be manually created.
+
+The Spring JdbcTemplate provides methods to write the queries directly, hence, saving time in writing such boilerplate code.
+
+Task | Spring | You
+-----|--------|----
+Connection Management | :far-check-circle: | 
+Writing SQL Queries | | :far-check-circle: 
+Statement Management | :far-check-circle: |
+ResultSet Management | :far-check-circle: |
+Row Data Retrieval | | :far-check-circle: 
+Parameter Declaration | | :far-check-circle: 
+Parameter Setting | :far-check-circle: |
+Transaction Management | :far-check-circle: |
+Exception Handling | :far-check-circle: |
+
+### Spring AOP
+
+Spring Framework addresses
+<tooltip content="Concerns that can affect the whole application and should be centralized in one location in code as much as possible. Examples of these concerns include transaction management, authentication, logging, security etc.">
+<i>cross-cutting concerns</i></tooltip> by supporting Aspect Oriented Programming (AOP).
+It provides ways to dynamically add the cross-cutting concerns before, after or around the actual logic using simple
+pluggable configurations. 
+
+For example, say we wish to log something into the console everytime before we call the method `getEmployeeById`.
+We can then write aspect class annotated with `@Aspect` annotation and write point-cut expressions to log program information from
+<tooltip content="Join point is a point of execution of the program, such as the execution of a method or the handling of an exception.">
+<i>joint-point</i>
+</tooltip> methods.
+```java
+@Aspect
+public class EmployeeCRUDAspect {
+      
+    @Before("execution(* EmployeeManager.getEmployeeById(..))") // point-cut expressions
+    public void logBefore(JoinPoint joinPoint) {
+        System.out.println("EmployeeCRUDAspect.logBefore() : "
+                + joinPoint.getSignature().getName());
+    }
+    
+    @After("execution(* EmployeeManager.getEmployeeById(..))") // point-cut expressions
+    public void logAfter(JoinPoint joinPoint) {
+        System.out.println("EmployeeCRUDAspect.logAfter() : "
+                + joinPoint.getSignature().getName());
+    }
+}
+```
+    
+And the corresponding `EmployeeManager.java` file could be:
+```java
+@Component
+public class EmployeeManager
+{
+    public EmployeeDTO getEmployeeById(Integer employeeId) {
+        System.out.println("Method getEmployeeById() called");
+        return new EmployeeDTO();
+    }
+}
+```
```

Comment:
Use Egyptian style braces?
Also, if you took these code examples from somewhere, cite the source.